### ISSUE TITLE
Ctrl + Z removes entire filter instead of rolling back latest change

### ISSUE THREAD

**Issue by TheBlackEntity** [jliljebl/flowblade#1154]
Please include the following information.
1) Flowblade version (Help->About): 2.12.0.2
2) MLT version (Help->Runtime Environment): 7.22.0
3) Your distribution and desktop environment(Ubuntu/Unity, Debian/KDE, Mint/Cinnamon etc.): NixOS, KDE
4) X11 or Wayland: X11
5) If possible, try to provide step-by-step instructions on how to reproduce the issue. From developer point of view this is the easieast way to understand what is happening.:
Add a filter (For example, transform), make multiple changes, hit Ctrl + Z.

Using Ctrl + Z removes the entire filter instead of rolling back the latest change I made to the filter.

Is this intended?

I personally think of it as extremely annoying.

If I make a small mistake and just want to roll it back, I have to do it manually and cannot just use Ctrl + Z, since that just removes the last filter I added.

https://github.com/jliljebl/flowblade/assets/36412486/0e4b99b2-a38a-4ff9-8acd-34c480368389

**Comment by jliljebl** [comment]
> Is this intended?

Yes. I made a decision a long time ago that undo system only takes into account timeline structural changes and parameter values are not part of the system.

I have personally never felt this omission to be a problem, and there has not been requests for it to be included.

However, I have come around to change some other earlier decisions, and this could be looked into. The problem with this one is that it is not completely clear how destabilizing and difficult this is to do, so I don't have a clear answer on if and when we can do this. I'll do exploratory work on his during next development cycle, and update here on prospects of doing this.

**Comment by TheBlackEntity** [comment]
Thank you for looking into it.

I really hope this is going to be added, since that's the only dealbreaker with this software.

The rest is quite intuitive and well made :)

**Comment by TheBlackEntity** [comment]
Hey!

Just wanted to ask if there are any updates on this? :3

**Comment by jliljebl** [comment]
I'm doing the next release late June or more likely sometime in August (July is always quite low activity due summer holidays in Finland etc.). 

I always do the features I'm planning to do for the release before doing the kind of work that the test branch I'm planning to do for this. 

Most likely work on this will happen mid-June or early August. I'll post update after i have the information I'm planning the discover by doing work on test branch that may turn into feature branch.
